sys/posix/sockets/posix_sockets.c

```diff
@@ -1144,7 +1144,8 @@ int setsockopt(int socket, int level, int option_name, const void *option_value,
 
     tv = (struct timeval *) option_value;
 
-    if (tv->tv_sec < 0 || tv->tv_usec < 0) {
+    /* tv_sec is unit32_t, so never negative */
```

Comment:
Are you sure? According to [the standard](https://pubs.opengroup.org/onlinepubs/7908799/xsh/systime.h.html), it's `time_t`, which is usually `int64_t` these days.

Reply:
I agree that the standard refers to `time_t`, and the 3 definitions of `tv_sec `all refer to `time_t `in RIOT as per
````
  # pri kind tag               file
  1 F C m    tv_sec            ./cpu/esp8266/include/sys/types.h
               struct:timespec
               time_t  tv_sec;   /* Seconds */
  2 F   m    tv_sec            ./cpu/avr8_common/avr_libc_extra/include/sys/time.h
               struct:timeval
               time_t         tv_sec;      /**< seconds */
  3 F   m    tv_sec            ./cpu/avr8_common/avr_libc_extra/include/vendor/time.h
               struct:timespec
               time_t    tv_sec;
````
However, in RIOT, in the 3 places where `time_t `is defined we have
````
  # pri kind tag               file
  1 F C t    time_t            ./cpu/esp8266/include/sys/types.h
               typedef _TIME_T_ time_t;
  2 F   t    time_t            ./cpu/avr8_common/avr_libc_extra/include/sys/types.h
               typedef     uint32_t time_t;      /**< Used for time in seconds */
  3 F   t    time_t            ./cpu/avr8_common/avr_libc_extra/include/vendor/time.h
               typedef uint32_t time_t;
````
but `_TIME_T_ `is not defined anywhere.

So it looks like this change will have to be some sort of conditional based on an AVR build, but haven't worked out which conditional to use yet.

Reply:
Ah so you are getting a build warning (that gets promited to error) on AVR with this check in place?

Reply:
I'd be fine with dropping that check, it might as well have been an `assert()`.